app/models/service_binding.rb

```diff
@@ -28,34 +28,55 @@ def create(app_id)
 
     raise RoleAlreadyCreated.new("Host identity already exists.") if host.exists?
 
-    result = load_policy(template_create)
-    host =
+    host_id =
       if ConjurClient.policy != 'root'
-        "host/#{ConjurClient.policy}/#{@binding_id}"
+        "#{ConjurClient.policy}/#{@binding_id}"
       else
-        "host/#{@binding_id}"
+        @binding_id
       end
     
+    api_key = (ConjurClient.version == 4 ? create_v4(host_id) : create_v5)
+
     return {
       account: ConjurClient.account,
       appliance_url: ConjurClient.appliance_url,
-      authn_login: host,
-      authn_api_key: result.created_roles.values.first['api_key'],
+      authn_login: "host/#{host_id}",
+      authn_api_key: api_key,
       ssl_certificate: ConjurClient.ssl_cert || "",
       version: ConjurClient.version
     }
   end
 
   def delete
     host = conjur_api.role(role_name)
+    
     raise HostNotFound if !host.exists?
 
     host.rotate_api_key
-    load_policy template_delete, method: Conjur::API::POLICY_METHOD_PATCH
+
+    if ConjurClient.version == 5
+      load_policy template_delete, method: Conjur::API::POLICY_METHOD_PATCH
+    end
   end
 
   private
 
+  def create_v4(host_id)
+    hf_token =
+      conjur_api.
+        resource("cucumber:host_factory:pcf%2Fpcf-apps").
```

Comment:
the first `pcf` here should be `ConjurClient.policy`, and `cucumber` should be `ConjurClient.account`

Reply:
Oops, forgot to clean this up. Will do so.